src/stores/TransactionStore.js

```diff
@@ -3,18 +3,60 @@ import axios from 'axios';
 import { map, includes, isEmpty, remove, each, reduce, findIndex, cloneDeep } from 'lodash';
 import { WalletProvider, TransactionType, TransactionStatus, Token, TransactionGas } from 'constants';
 import { Transaction, TransactionCost } from 'models';
+import { AbiCoder } from 'web3-eth-abi';
+import promisify from 'js-promisify';
 
 import networkRoutes from '../network/routes';
 import { queryAllTransactions } from '../network/graphql/queries';
 import { createTransaction } from '../network/graphql/mutations';
 import getContracts from '../config/contracts';
 import Tracking from '../helpers/mixpanelUtil';
 
+const web3EthAbi = new AbiCoder();
 const INIT_VALUES = {
   visible: false,
   provider: WalletProvider.NAKA,
 };
 
+const CREATE_EVENT_FUNC_SIG = '2b2601bf';
+const BET_EVENT_FUNC_SIG = '885ab66d';
+const SET_EVENT_FUNC_SIG = 'a6b4218b';
+const VOTE_EVENT_FUNC_SIG = '1e00eb7f';
+const createEventFuncTypes = [
+  'string',
+  'bytes32[10]',
+  'uint256',
+  'uint256',
+  'uint256',
+  'uint256',
+  'address',
+];
+const playEventFuncTypes = [
```

Comment:
lets be consistent and call it betting instead of playing so that if anyone works on both the back/front end they know what is what.

Reply:
We have bet, vote, setresult using the same `playEventFuncTypes` and same `playEvent function`, Do we need to reuse the same code? think it may be cleaner. If we separate  them into different functions, it may cause 3 times duplicate code.

Reply:
play is fine i guess? or `participate`?